docs/ibc/light-clients/overview.md

```diff
@@ -1,3 +1,45 @@
 <!--
 order: 1
--->
\ No newline at end of file
+-->
+
+# Overview 
+
+Learn how to build IBC lightclient modules and fulfill the interfaces required to integrate with core IBC. {synopsis}
+
+The following aims to provide a high level IBC lightclient module developer guide. The lifecycle of IBC lightclients are created and maintained by the core IBC `MsgServer` and the abstractions provided by the `02-client` submodule. A lightclient module developer is only required to implement the methods and fulfil the expectations of a defined set of interfaces. In ibc-go these interfaces reside within the `core/modules/exported` package. 
+
+Throughout this guide the `07-tendermint` lightclient module may be referred to as a reference example.
+
+## Concepts and vocabulary
+
+This guide serves to be implementation specific with respect to ibc-go. Readers are expected to familiarize themselves with the IBC protocol specifications.
+Please refer to the defintions outlined in [ICS-002 Client Semantics](https://github.com/cosmos/ibc/tree/main/spec/core/ics-002-client-semantics#Definitions).
+
+### ClientState 
+
+ClientState is a term used to define the data structure which encapsulates opaque lightclient state. This refers to internal data that accommodate the semantics of a lightclient. This may be any arbitrary data such as:
+
+- Constraints used for client updates.
+- Constraints used for misbehaviour detection.
+- Constraints used for state verification.
+- Constraints used for client upgrades.
+
```

Comment:
Does it make sense here to mention that some properties are customizable, and others are fixed during the client lifecycle?

Reply:
After taking a browse through some of the other existing docs I've noticed fields for tendermint client upgrades are documented [here](https://ibc.cosmos.network/main/ibc/upgrades/quick-guide.html#ibc-client-breaking-upgrades). I think we can focus in on mutability of client state fields in the dedicated client upgrades page (upgrades.md) and link to the existing upgrade docs for tm clients there also